src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs

```diff
@@ -1101,40 +1101,90 @@ public async Task Pound_r_nuget_works_immediately_after_a_language_selector(Lang
         [Theory]
         [InlineData(Language.CSharp)]
         [InlineData(Language.FSharp)]
-        public async Task Pound_r_nuget_should_display_only_requested_packages(Language defaultLanguageKernel)
+        public async Task Pound_r_nuget_should_display_only_requested_packages_for_first_submission(Language defaultLanguageKernel)
         {
             var kernel = CreateCompositeKernel(defaultLanguageKernel);
 
             var code = @"
-#r ""nuget:Google.Protobuf""
 #r ""nuget:Microsoft.ML.OnnxTransformer,1.4.0""
 ";
 
             var command = new SubmitCode(code);
-
             var result = await kernel.SendAsync(command);
-
             using var events = result.KernelEvents.ToSubscribedList();
 
             using var _ = new AssertionScope();
 
             var expectedDisplayed = new[]
             {
-                "Google.Protobuf, 3.17.3",
                 "Microsoft.ML.OnnxTransformer, 1.4.0"
             };
 
             var notExpectedDisplayed = new[]
             {
-                "Microsoft.ML.KMeansClustering",
-                "Microsoft.ML.PCA",
-                "Microsoft.ML.StandardTrainers",
-                "Microsoft.ML.Transforms",
-                "Microsoft.ML.CpuMath",
-                "Microsoft.ML.DataView",
-                "Microsoft.ML.OnnxRuntime",
-                "Newtonsoft.Json",
-                "System.CodeDom"
+                "google.protobuf, 3.5.1",
+                "microsoft.ml.kmeansclustering",
+                "microsoft.ml.pca",
+                "microsoft.ml.standardtrainers",
+                "microsoft.ml.transforms",
+                "microsoft.ml.cpumath",
+                "microsoft.ml.dataview",
+                "microsoft.ml.onnxruntime",
+                "newtonsoft.json",
+                "system.codedom"
+            };
+
+            events.OfType<DisplayedValueUpdated>()
+                  .Where(v => v.Value is InstallPackagesMessage)
+                  .Last().Value
+                  .As<InstallPackagesMessage>()
+                  .InstalledPackages
+                  .Aggregate((s, acc) => acc + " & " + s)
+                  .Should()
+                  .ContainAll(expectedDisplayed)
```

Comment:
You can use `BeEquivalentTo` without the aggregation, which will be more concise and provide better error output. This will also avoid having to specify the packages you don't expect to see.

Reply:
Updated